contracts/protocol/v1/proxies/P1LiquidatorProxy.sol

```diff
@@ -0,0 +1,282 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.16;
+pragma experimental ABIEncoderV2;
+
+import { Ownable } from "@openzeppelin/contracts/ownership/Ownable.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import { BaseMath } from "../../lib/BaseMath.sol";
+import { SignedMath } from "../../lib/SignedMath.sol";
+import { I_PerpetualV1 } from "../intf/I_PerpetualV1.sol";
+import { P1BalanceMath } from "../lib/P1BalanceMath.sol";
+import { P1Types } from "../lib/P1Types.sol";
+
+
+/**
+ * @title P1LiquidatorProxy
+ * @author dYdX
+ *
+ * @notice Proxy contract allowing users to liquidate accounts and pay some extra fees to the
+ * insurance fund.
+ */
+contract P1LiquidatorProxy is
+    Ownable
+{
+    using BaseMath for uint256;
+    using SignedMath for SignedMath.Int;
+    using P1BalanceMath for P1Types.Balance;
+
+    // ============ Events ============
+
+    event LogLiquidatorProxyUsed(
+        address indexed liquidator,
+        address indexed liquidatee,
+        bool isBuy,
+        uint256 liquidationAmount,
+        uint256 feeAmount
+    );
+
+    event LogInsuranceFundSet(
+        address insuranceFund
+    );
+
+    // ============ Immutable Storage ============
+
+    // address of the perpetual contract
+    address public _PERPETUAL_V1_;
+
+    // address of the liquidator contract
+    address public _LIQUIDATOR_;
+
+    // address of the insurance fund
+    address public _INSURANCE_FUND_;
+
+    // base number representing the percentage of fees that go to the insurance fund
+    uint256 public _FEE_PERCENTAGE_;
+
+    // ============ Constructor ============
+
+    constructor (
+        address perpetualV1,
+        address liquidator,
+        address insuranceFund,
+        uint256 feePercentage
+    )
+        public
+    {
+        _PERPETUAL_V1_ = perpetualV1;
+        _LIQUIDATOR_ = liquidator;
+        _INSURANCE_FUND_ = insuranceFund;
+        _FEE_PERCENTAGE_ = feePercentage;
+    }
+
+    // ============ External Functions ============
+
+    /**
+     * @notice Sets infinite allowance on the perpetual contract.
+     * @dev Cannot be run in the constructor due to technical restrictions in Solidity.
+     */
+    function setAllowance()
+        external
+    {
+        address perpetual = _PERPETUAL_V1_;
+        address tokenContract = I_PerpetualV1(perpetual).getTokenContract();
+        SafeERC20.safeApprove(
+            IERC20(tokenContract),
+            perpetual,
+            uint256(-1)
+        );
+    }
+
+    /**
+     * @notice Allows an account below the minimum collateralization to be liquidated by another
+     *  account. This allows the account to be partially or fully subsumed by the liquidator.
```

Comment:
Might be a good place to note that the caller must authorize this liquidator proxy contract as a local operator on their perpetual account before calling this function

Reply:
We can just make it a global operator